cmd/add.go

```diff
@@ -45,30 +55,39 @@ func add(cmd *cobra.Command, args []string) error {
 		})
 	}
 
-	// Ask username password from prompt
-	organization, err := lib.Prompt("Okta organization", false)
-	if err != nil {
-		return err
+	// Ask Okta organization details and username if not given in command line arguments
+	if organization == "" {
+		organization, err = lib.Prompt("Okta organization", false)
+		if err != nil {
+			return err
+		}
 	}
 
-	oktaRegion, err := lib.Prompt("Okta region ([us], emea, preview)", false)
-	if err != nil {
-		return err
+	if oktaRegion == "" {
+		oktaRegion, err = lib.Prompt("Okta region ([us], emea, preview)", false)
+		if err != nil {
+			return err
+		}
 	}
 	if oktaRegion == "" {
 		oktaRegion = "us"
 	}
 
-	oktaDomain, err := lib.Prompt("Okta domain ["+oktaRegion+".okta.com]", false)
-	if err != nil {
-		return err
+	if oktaDomain == "" {
+		oktaDomain, err = lib.Prompt("Okta domain ["+organization+".okta.com]", false)
```

Comment:
I've been meaning to look into this. I think the current implementation is incorrect. I'm pretty sure it should be something like

```
tld := lib.GetOktaDomain(region)
fmt.Sprintf("[%s.%s]", organization, tld)
```

getOktaDomain is in `lib` and is not exported; I'd be fine if you exported it.

Reply:
you are right, the prompt never showed the right `tld` if the user is not picking the us region. The code resolves it fine in the end if left blank, so it's just a UI fix.